### PR TITLE
WebContextMenus: implement context menu for text selection

### PR DESCRIPTION
**Pull request by Masterjoona**
add search with google, copy words and automod filter words to web message context menu

### CODE REVIEW

File: src/plugins/webContextMenus.web/index.ts [hunk 1]
```diff
@@ -200,6 +200,28 @@ export default definePlugin({
...
+            replacement: {
+                match: /\i\.isPlatformEmbedded/,
+                replace: "true"
+            }
+        },
+        {
+            find: "focus()}}):null",
+            replacement: {
+                match: /\i\.isPlatformEmbedded/,
+                replace: "true"
```
Comment (Vendicated) [inline src/plugins/webContextMenus.web/index.ts:215]: this module uses DiscordNative copy which makes copying not work

image

File: src/plugins/webContextMenus.web/index.ts [hunk 2]
```diff
@@ -200,6 +200,28 @@ export default definePlugin({
...
+            }
+        },
+        // Automod add filter words
+        {
+            find: "({contextMenu:!0}),",
+            replacement: {
+                match: /\i\.isPlatformEmbedded/,
```
Comment (Vendicated) [inline src/plugins/webContextMenus.web/index.ts:222]:
```suggestion
        {
            find: '("interactionUsernameProfile',
            replacement: {
                match: /\i\.isPlatformEmbedded(?=.{0,50}\.tagName)/,
```

File: src/plugins/webContextMenus.web/index.ts [hunk 3]
```diff
@@ -200,6 +200,28 @@ export default definePlugin({
...
+        },
+        {
+            find: ".Messages.SEARCH_WITH_GOOGLE",
+            replacement: {
+                match: /\i\.isPlatformEmbedded/,
+                replace: "true"
+            }
+        },
+        {
+            find: "focus()}}):null",
```
[the remark this answers is not in the document]
Comment (Masterjoona) [reply src/plugins/webContextMenus.web/index.ts:212]: is `.Messages.COPY,hint:` fine`?

[the remark this answers is not in the document]
Comment (Vendicated) [reply src/plugins/webContextMenus.web/index.ts:212]: if it's unique

File: src/plugins/webContextMenus.web/index.ts (hunk 1 above)
Comment (Masterjoona) [reply src/plugins/webContextMenus.web/index.ts:215]: oops how did i not realise such an obvious thing

File: src/plugins/webContextMenus.web/index.ts (hunk 2 above)
Comment (Masterjoona) [reply src/plugins/webContextMenus.web/index.ts:222]: why the lookahead? Future-proofing?

Comment (Vendicated) [reply src/plugins/webContextMenus.web/index.ts:222]: to make the match more generic to ensure you won't match unrelated stuff that might be added in the future or that could false positive your find

**Comment by Vendicated** [comment]
thanks!

**Pull request merged**